### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by aleksandryakunichev** [jpalardy/vim-slime#204]
Hi! Thanks a lot for this awesome plugin, but I'm struggling with code execution performance :( 

I'm using `vim-slime` for GHC, but sending code to tmux's pane is really slow — about 3 seconds. Here is the video:

![vim tidal performance issue](https://res.cloudinary.com/canvaskisa/image/upload/v1561116156/vim-tidal_y3e5fi.gif "vim tidal performance issue")

And profile log:
```
FUNCTIONS SORTED ON TOTAL TIME
count  total (s)   self (s)  function
    1   2.162968   0.051221  slime#send_op()
    1   2.111711   0.000081  slime#send()
    1   2.111596   0.000069  <SNR>126_SlimeDispatch()
    1   2.111527   0.000136  <SNR>126_TmuxSend()
    2   1.361202   0.000114  <SNR>126_TmuxCommand()
    1   0.750190   0.000027  <SNR>126_WritePasteFile()
    1   0.051045   0.000047  provider#clipboard#Call()

```
I'm really sad with this issue :( [here is](https://github.com/aleksandryakunichev/.vimrc/blob/master/init.vim#L112) my config if it can help!

**Comment by jpalardy** [comment]
Hi @aleksandryakunichev 

Whoah, that is definitely my experience with slime...

Some questions:
- is it the same result with other REPLs?
- is it the same result with a minimal vim config? (just vim-slime)
- what version of vim, tmux, and OS is this?
- how did you get this profile log? :smile: am I reading correctly or the `self` indicates the time isn't actually spent in vim?

Let me know.

**Comment by aleksandryakunichev** [comment]
Thanks for a quick answer!

* is it the same result with other REPLs?

I'm using it along with `tmux` eq `let g:slime_target = "tmux"` and for every repl it is the same, unfortunately.

* is it the same result with a minimal vim config? (just vim-slime)

Yeah :( I've removed all plugins, mappings — leaving only `vim-slime` with default mappings and `Plug`, but the result is the same.

* what version of vim, tmux, and OS is this?

`nvim -v` — `NVIM v0.3.7`
`tmux -V` — `tmux 2.9a`
`macOS Mojave 10.14.5 (18F132)`

* how did you get this profile log? 😄 am I reading correctly or the self indicates the time isn't actually spent in vim?

I've used profiling tutorial from [this article](https://www.systutorials.com/136414/profiling-vim-to-find-out-which-plugin-makes-vim-slow/). I ran `:profile start profile.log` and other startup commands — then executed `vim-slime` and stopped the profiling. And you're totally right about `self` :)

**Comment by jpalardy** [comment]
Hmmmm ... nothing obvious pops at me :thinking: 

Maybe it's tmux, would you be willing to try GNU screen and see what happens?

**Comment by aleksandryakunichev** [comment]
@jpalardy After a couple of hours, I've found the real problem with this issue.

My shell was `fish` and it's startup time "wished me best" (it was about ~2s) and `nvim` used this shell as a default, running all commands in `fish`. 

The soultion for speeding up `vim-slime` performance was to set up shell directly in `.vimrc`, so it will use `sh`, `bash` or whatever. So I did it:

```vim
set shell=/bin/sh " set default shell
```

And now everything is insanely fast. 

Thanks a lot for your help! 🙌
